lib/loc/location.rb

```diff
@@ -3,6 +3,7 @@ class Location
     attr_reader :lat, :lng
 
     def initialize(lat, lng)
+      raise TypeError unless lat.is_a?(Numeric) && lng.is_a?(Numeric)
```

Comment:
Peux-tu raise quelquechose de plus précis?

En ruby, le message classique est:

```
TypeError (wrong argument type Float (expected String))
```

Ici on pourrait même se permettre de dire:

```
lat should be a Numeric
```
non?

Et pourquoi pas utiliser la méthode `Float()`. Ça accepte certaines chaines de caractères très précises, et il y a un message d'erreur très clair. Bon c'est vrai que du coup ça accepte les chaines de caractères... alors peut-être pas. À toi de pour cette part! Par contre le message est important ;)

Reply:
You are right, it should be a custom message.
For `Float()` I am not sure there is a valid case where geopositions could be string.